ResearchKit/Common/ORKAnswerFormat.m

```diff
@@ -2062,6 +2071,66 @@ - (ORKAnswerFormat *)impliedAnswerFormat {
 @end
 
 
+#pragma mark - ORKConfirmTextAnswerFormat
+
+@implementation ORKConfirmTextAnswerFormat
+
+- (Class)questionResultClass {
+    return [ORKBooleanQuestionResult class];
+}
+
+- (instancetype)initWithOriginalItemIdentifier:(NSString *)originalItemIdentifier {
+    self = [super init];
+    if (self) {
+        _originalItemIdentifier = [originalItemIdentifier copy];
+    }
+    return self;
+}
+
+- (BOOL)isAnswerValid:(id)answer {
+    return [(NSNumber *)answer isEqual:@YES] ? YES : NO;
```

Comment:
I would check `isKindOfClass` before comparing. And do we need to check  `ORKIsAnswerEmpty()`? I am not sure.